integration-cli/docker_cli_restart_test.go

```diff
@@ -275,4 +275,7 @@ func (s *DockerSuite) TestRestartAutoRemoveContainer(c *check.C) {
 
 	out, _ = dockerCmd(c, "ps")
 	c.Assert(out, checker.Contains, id[:12], check.Commentf("container should be restarted instead of removed: %v", out))
+
+	// Kill the container to make sure it will be removed
+	dockerCmd(c, "kill", id)
```

Comment:
So, this tests is making sure a container started with `--rm` will not be removed when doing a restart one it. This means, when `TearDown` gets there, it's failing with `Removal in progress`. By killing it before, I make sure it's getting cleaned before.